lib/src/convert.dart

```diff
@@ -20,7 +22,7 @@ const StringCodec stringCodec = StringCodec();
 abstract class DataCodec<T> extends Codec<T, List<int>> {
   static final List<DataCodec<Object>> codecs =
       List<DataCodec<Object>>.unmodifiable(
-          <DataCodec<Object>>[blobCodec, intCodec, floatCodec, stringCodec]);
+          <DataCodec<Object>>[blobCodec, intCodec, floatCodec, stringCodec, doubleCodec]);
```

Comment:
Could you preserve the sorting here? Something like:

```dart
<DataCodec<Object>>[blobCodec, doubleCodec, intCodec, floatCodec, stringCodec]);
```

Reply:
I originally considered this, but my thinking is to move the extended types like double to the back of the list. 

For decoding there's no issue, since that would be dependent on an external API. 

For encoding, since doubles are outside of the core recommended types, I want to preserve floats being checked first, and only intentionally encode doubles.

Reply:
Fantastic. That rationale makes perfect sense. Could you add a comment to this effect so it doesn't get (wrongly) sorted down the road?